Ofborg says
```
Compiling /tmp/nix-build-milkytracker-1.04.00.drv-0/source/src/tracker/cocoa/resources/Application.xib
xcode-select: error: no developer tools were found at '/Applications/Xcode.app', and no install could be requested (perhaps no UI is present), please install manually from 'developer.apple.com'.
make[2]: *** [src/tracker/CMakeFiles/tracker.dir/build.make:1517: src/tracker/MilkyTracker.app/Contents/MacOS/MilkyTracker] Error 1
```
So back to Linux-only we go. :cry:

--- a/pkgs/applications/audio/milkytracker/default.nix
+++ b/pkgs/applications/audio/milkytracker/default.nix
@@ -47,19 +47,6 @@ stdenv.mkDerivation (finalAttrs: {
     pkg-config
   ];
 
-  # Requires `ibtool`, but the version in `xib2nib` is not capable of
-  # building these apps. Build them using `ibtool` from Xcode, but don't allow any other binaries
-  # into the sandbox. Note that the CLT are not supported because `ibtool` requires Xcode.
-  sandboxProfile = lib.optionalString stdenv.isDarwin ''
-    (allow process-exec
-      (literal "/usr/bin/ibtool")
-      (regex "/Xcode.app/Contents/Developer/usr/bin/ibtool")
-      (regex "/Xcode.app/Contents/Developer/usr/bin/xcodebuild"))
-    (allow file-read*)
-    (deny file-read* (subpath "/usr/local") (with no-log))
-    (allow file-write* (subpath "/private/var/folders"))
-  '';
-
   buildInputs = [
     lhasa
     libjack2
@@ -79,8 +66,6 @@ stdenv.mkDerivation (finalAttrs: {
   cmakeFlags = [
     # Somehow this does not get set automatically
     "-DSDL2MAIN_LIBRARY=${SDL2}/lib/libSDL2${stdenv.hostPlatform.extensions.sharedLibrary}"
-  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
-    "-DIBTOOL=/usr/bin/ibtool"
   ];
 
   postInstall = lib.optionalString stdenv.hostPlatform.isLinux ''
@@ -98,6 +83,8 @@ stdenv.mkDerivation (finalAttrs: {
     homepage = "https://milkytracker.org/";
     license = licenses.gpl3Plus;
     platforms = platforms.unix;
+    # ibtool -> real Xcode -> I can't get that, and Ofborg can't test that
+    broken = stdenv.hostPlatform.isDarwin;
     maintainers = with maintainers; [ OPNA2608 ];
   };
 })

--- a/pkgs/applications/audio/milkytracker/default.nix
+++ b/pkgs/applications/audio/milkytracker/default.nix
@@ -1,37 +1,62 @@
-{ lib, stdenv, fetchFromGitHub, fetchpatch, cmake, pkg-config, makeWrapper
-, SDL2, alsa-lib, libjack2, lhasa, perl, rtmidi, zlib, zziplib }:
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, gitUpdater
+, cmake
+, pkg-config
+, makeWrapper
+, SDL2
+, alsa-lib
+, libjack2
+, lhasa
+, perl
+, rtmidi
+, zlib
+, zziplib
+}:
 
-stdenv.mkDerivation rec {
-  version = "1.03.00";
+stdenv.mkDerivation (finalAttrs: {
   pname = "milkytracker";
+  version = "1.04.00";
 
   src = fetchFromGitHub {
-    owner  = "milkytracker";
-    repo   = "MilkyTracker";
-    rev    = "v${version}";
-    sha256 = "025fj34gq2kmkpwcswcyx7wdxb89vm944dh685zi4bxx0hz16vvk";
+    owner = "milkytracker";
+    repo = "MilkyTracker";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-ta4eV/FGBfgTppJwDam0OKQ7udtlinbWly/FPCE+Qss=";
   };
 
   patches = [
+    # Fix crash after querying midi ports
+    # Remove when version > 1.04.00
     (fetchpatch {
-      name = "CVE-2022-34927.patch";
-      url = "https://github.com/milkytracker/MilkyTracker/commit/3a5474f9102cbdc10fbd9e7b1b2c8d3f3f45d91b.patch";
-      hash = "sha256-YnN1Khcbct7iG7TdwxFU1XVCeKR/Zrhe+oMepvh8cRU=";
+      url = "https://github.com/milkytracker/MilkyTracker/commit/7e9171488fc47ad2de646a4536794fda21e7303d.patch";
+      hash = "sha256-CmnIwmGGnsnlRrvVAXe2zaQf1CFMB5BJPKmiwGOHgGY=";
     })
   ];
 
-  postPatch = ''
-    # https://github.com/milkytracker/MilkyTracker/issues/262
-    substituteInPlace CMakeLists.txt \
-      --replace 'CMAKE_CXX_STANDARD 98' 'CMAKE_CXX_STANDARD 11'
-  '';
+  nativeBuildInputs = [
+    cmake
+    makeWrapper
+    pkg-config
+  ];
 
-  nativeBuildInputs = [ cmake pkg-config makeWrapper ];
+  buildInputs = [
+    alsa-lib
+    lhasa
+    libjack2
+    perl
+    rtmidi
+    SDL2
+    zlib
+    zziplib
+  ];
 
-  buildInputs = [ SDL2 alsa-lib libjack2 lhasa perl rtmidi zlib zziplib ];
-
-  # Somehow this does not get set automatically
-  cmakeFlags = [ "-DSDL2MAIN_LIBRARY=${SDL2}/lib/libSDL2.so" ];
+  cmakeFlags = [
+    # Somehow this does not get set automatically
+    "-DSDL2MAIN_LIBRARY=${SDL2}/lib/libSDL2.so"
+  ];
 
   postInstall = ''
     install -Dm644 $src/resources/milkytracker.desktop $out/share/applications/milkytracker.desktop
@@ -39,11 +64,15 @@ stdenv.mkDerivation rec {
     install -Dm644 $src/resources/milkytracker.appdata $out/share/appdata/milkytracker.appdata.xml
   '';
 
+  passthru.updateScript = gitUpdater {
+    rev-prefix = "v";
+  };
+
   meta = with lib; {
     description = "Music tracker application, similar to Fasttracker II";
     homepage = "https://milkytracker.org/";
     license = licenses.gpl3Plus;
-    platforms = [ "x86_64-linux" "i686-linux" ];
-    maintainers = with maintainers; [];
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ OPNA2608 ];
   };
-}
+})

--- a/pkgs/applications/audio/milkytracker/default.nix
+++ b/pkgs/applications/audio/milkytracker/default.nix
@@ -3,15 +3,18 @@
 , fetchFromGitHub
 , fetchpatch
 , gitUpdater
-, cmake
-, pkg-config
-, makeWrapper
-, SDL2
 , alsa-lib
+, cmake
+, Cocoa
+, CoreAudio
+, Foundation
 , libjack2
 , lhasa
+, makeWrapper
 , perl
+, pkg-config
 , rtmidi
+, SDL2
 , zlib
 , zziplib
 }:
@@ -36,14 +39,28 @@ stdenv.mkDerivation (finalAttrs: {
     })
   ];
 
+  strictDeps = true;
+
   nativeBuildInputs = [
     cmake
     makeWrapper
     pkg-config
   ];
 
+  # Requires `ibtool`, but the version in `xib2nib` is not capable of
+  # building these apps. Build them using `ibtool` from Xcode, but don't allow any other binaries
+  # into the sandbox. Note that the CLT are not supported because `ibtool` requires Xcode.
+  sandboxProfile = lib.optionalString stdenv.isDarwin ''
+    (allow process-exec
+      (literal "/usr/bin/ibtool")
+      (regex "/Xcode.app/Contents/Developer/usr/bin/ibtool")
+      (regex "/Xcode.app/Contents/Developer/usr/bin/xcodebuild"))
+    (allow file-read*)
+    (deny file-read* (subpath "/usr/local") (with no-log))
+    (allow file-write* (subpath "/private/var/folders"))
+  '';
+
   buildInputs = [
-    alsa-lib
     lhasa
     libjack2
     perl
@@ -51,14 +68,22 @@ stdenv.mkDerivation (finalAttrs: {
     SDL2
     zlib
     zziplib
+  ] ++ lib.optionals stdenv.hostPlatform.isLinux [
+    alsa-lib
+  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    Cocoa
+    CoreAudio
+    Foundation
   ];
 
   cmakeFlags = [
     # Somehow this does not get set automatically
-    "-DSDL2MAIN_LIBRARY=${SDL2}/lib/libSDL2.so"
+    "-DSDL2MAIN_LIBRARY=${SDL2}/lib/libSDL2${stdenv.hostPlatform.extensions.sharedLibrary}"
+  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    "-DIBTOOL=/usr/bin/ibtool"
   ];
 
-  postInstall = ''
+  postInstall = lib.optionalString stdenv.hostPlatform.isLinux ''
     install -Dm644 $src/resources/milkytracker.desktop $out/share/applications/milkytracker.desktop
     install -Dm644 $src/resources/pictures/carton.png $out/share/pixmaps/milkytracker.png
     install -Dm644 $src/resources/milkytracker.appdata $out/share/appdata/milkytracker.appdata.xml
@@ -72,7 +97,7 @@ stdenv.mkDerivation (finalAttrs: {
     description = "Music tracker application, similar to Fasttracker II";
     homepage = "https://milkytracker.org/";
     license = licenses.gpl3Plus;
-    platforms = platforms.linux;
+    platforms = platforms.unix;
     maintainers = with maintainers; [ OPNA2608 ];
   };
 })

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -29933,7 +29933,9 @@ with pkgs;
 
   cutecapture = callPackage ../applications/video/cutecapture { };
 
-  milkytracker = callPackage ../applications/audio/milkytracker { };
+  milkytracker = callPackage ../applications/audio/milkytracker {
+    inherit (darwin.apple_sdk.frameworks) Cocoa CoreAudio Foundation;
+  };
 
   ptcollab = libsForQt5.callPackage ../applications/audio/ptcollab { };
 

